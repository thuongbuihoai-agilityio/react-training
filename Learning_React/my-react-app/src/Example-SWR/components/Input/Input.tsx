import { ChangeEvent, memo, useState } from "react";

interface InputProps {
  addTodo: (title: string) => void;
}

const Input:React.FC<InputProps> = ({ addTodo }) => {
  const [title, setTitle] = useState("");

  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    setTitle(value);
  };

  const handleAddTodo = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.code === "Enter") {
      addTodo(title);
      setTitle("");
    }
  };

  return (
    <input
      placeholder="Add new todo"
      className="todo__input"
      onChange={onChange}
      onKeyPress={handleAddTodo}
      value={title}
    />
  );
};

export default memo(Input);
