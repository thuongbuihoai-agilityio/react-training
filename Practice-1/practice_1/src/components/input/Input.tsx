import React from "react";
import "./input.css"

interface InputProps {
  className: string;
  type: string;
  placeholder: string;
}

function Input({ className, type, placeholder }: InputProps): JSX.Element {
  return (
    <input className = {`input input--${className}`} type={type} placeholder={placeholder}/>
    )
  }
export { Input };