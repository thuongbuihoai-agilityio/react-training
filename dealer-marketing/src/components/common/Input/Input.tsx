import React from "react";
import styleInput from "./input.module.css";

interface placeholder {
  type: string;
  placeholder: string;
}

const Input: React.FC<placeholder> = ({ type, placeholder }) => {
  return (
    <input className={styleInput.input} type={type} placeholder={placeholder} />
  );
};

export default Input;
