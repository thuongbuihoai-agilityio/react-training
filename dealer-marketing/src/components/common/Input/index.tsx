import React from "react";
import styleInput from "./input.module.css";

interface InputProps {
  type: string;
  placeholder: string;
}

const Input: React.FC<InputProps> = ({
  type = "text",
  placeholder = "Your email...",
}) => (
  <input className={styleInput.input} type={type} placeholder={placeholder} />
);

export default Input;
