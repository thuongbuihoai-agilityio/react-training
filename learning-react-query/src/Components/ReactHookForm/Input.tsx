import { useForm } from "react-hook-form";

const Input = () => {
  const {
    register,
    handleSubmit,
    // Read the formState before render to subscribe the form state through the Proxy
    formState: { errors, isDirty, isSubmitting, touchedFields, submitCount },
  } = useForm();
  const onSubmit = (data: any) => console.log(data);

  console.log('check value', errors, isDirty, isSubmitting, touchedFields, submitCount);

  return (
    <>
    <h2>Check value</h2>
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("test")} />
      <input type="submit" />
    </form>
    </>
  );
}

export default Input;