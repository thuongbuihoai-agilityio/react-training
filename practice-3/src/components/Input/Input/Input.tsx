import React, { memo } from "react";
import { InputProps } from "@/types/input";

const Input: React.FC<InputProps> = ({
  id,
  onChange,
  className,
  type,
  name,
  value,
  min,
  multiple
}) => {

  return (
    <input data-testid="input"
      id={id}
      onChange={onChange}
      type={type}
      name={name}
      className={className}
      value={value}
      min={min}
      multiple={multiple}
    />
  );
}

export default memo(Input);
