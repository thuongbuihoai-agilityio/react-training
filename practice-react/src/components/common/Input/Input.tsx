import React, { memo } from "react";
import "./input.css";

interface placeholder {
  type: string;
  placeholder: string;
}

const Input: React.FC<placeholder> = memo(({ type, placeholder }) => {
  return <input className="input input__value" type={type} placeholder={placeholder} />;
});

export default Input;
