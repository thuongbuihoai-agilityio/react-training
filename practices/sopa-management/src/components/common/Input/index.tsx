import React, { memo } from 'react';

// Styles
import './input.css';

export enum InputType {
  default = '',
  info = 'info',
  error = 'error'
}

export enum InputTheme {
  default = '',
  info = 'info',
  error = 'error'
}

interface IconProps {
  id?: string;
  name?: string;
  hidden?: boolean;
  checked?: boolean;
  value?: string | number;
  label?: string;
  type?: string;
  style?: string;
  theme?: string;
  htmlFor?: string;
  classNameInput?: string;
  classNameLabel?: string;
  className?: string;
  onChange?: () => void;
  onBlur?: () => void;
}

const Input = React.forwardRef<HTMLInputElement, IconProps>(
  (
    {
      id = '',
      name = '',
      hidden = false,
      checked = false,
      value,
      label = '',
      type = '',
      theme = InputTheme.default,
      style = InputType.default,
      htmlFor = '',
      classNameInput = 'input',
      classNameLabel = 'label',
      className = '',
      onChange = () => {},
      onBlur = () => {}
    },
    ref
  ) => (
    <>
      {label ? (
        <div data-testid='input-value' className='input-wrapper'>
          <label
            htmlFor={htmlFor}
            className={`${classNameLabel} label-${theme}`}
          >
            {label}
          </label>
          <input
            id={id}
            ref={ref}
            name={name}
            type={type}
            className={`${classNameInput} input-${style}`}
            onChange={onChange}
            onBlur={onBlur}
          />
        </div>
      ) : (
        <input
          data-testid='input'
          id={id}
          name={name}
          hidden={hidden}
          defaultChecked={checked}
          type={type}
          value={value}
          aria-hidden='true'
          className={`${className} ${classNameInput} input-${style}`}
        />
      )}
    </>
  )
);

export default memo(Input);
