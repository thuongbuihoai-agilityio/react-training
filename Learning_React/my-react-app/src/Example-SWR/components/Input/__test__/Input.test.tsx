import { render } from "@testing-library/react";
import Input from "../Input"

describe("Input Component", () => {
  test("renders Input component", () => {
    render(<Input addTodo={() => {}} />);
  });
})