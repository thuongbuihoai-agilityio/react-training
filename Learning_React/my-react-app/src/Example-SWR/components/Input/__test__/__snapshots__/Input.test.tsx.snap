// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="todo__input"
    placeholder="Add new todo"
    value=""
  />
</DocumentFragment>
`;
